controller.go

```diff
@@ -902,7 +925,9 @@ addToStore:
 		c.Unlock()
 	}
 
-	c.arrangeUserFilterRule()
+	if c.hasIPTablesEnabled() {
```

Comment:
@phyber this flag is relevant to Linux only so it should be in `firewall_linux.go` . I see that `arrangeUserFilterRule` is called from multiple places (`service_linux.go`) so the logic should be part of `arrangeUserFilterRule`

Reply:
Can you please add a test-case for the hasIPTablesEnabled logic

Reply:
whatever service_linux.go invokes, its getting called for ProgramIngress. 
@phyber ,  are we trying to block any IP table rule creation by dockerd or only arrangeUserFilterRule()? 
I noticed there few other places in libnetwork we tried to create chain rule. so wanted to confirm  what is the expected outcome of this PR.

Reply:
We're trying to stop `dockerd` playing with any firewall rules at all, meaning the user must arrange any required rules themselves.

For a clearer example of my intent, my workaround for this breakage in Docker was to put an `iptables` script somewhere in the `PATH` that `dockerd` was running under. All this script did was `exit 0` so that `dockerd` would think it was achieving something and stop breaking the user's firewall configuration.

It's possible that there were more places to stop `dockerd` from creating rules, however, this location was the most obvious as these are created when `dockerd` starts up. I never got to see if it was creating more things further down the line because I had dumped the above script into `PATH` to makes it behave itself.

Why was this necessary? To stop Docker fighting with configuration management like Puppet, etc.

Reply:
As for moving the code around and testing, I'll be looking at that sometime on Monday.

Reply:
Thanks for your response.